pkg/cmd/pulumi/cmd/project_sdk_link.go

```diff
@@ -0,0 +1,816 @@
+	fmt.Println()
+	fmt.Println("     <dependencies>")
+	fmt.Println("         <dependency>")
+	fmt.Println("             <groupId>com.google.code.findbugs</groupId>")
+	fmt.Println("             <artifactId>jsr305</artifactId>")
+	fmt.Println("             <version>3.0.2</version>")
+	fmt.Println("         </dependency>")
+	fmt.Println("         <dependency>")
+	fmt.Println("             <groupId>com.google.code.gson</groupId>")
+	fmt.Println("             <artifactId>gson</artifactId>")
+	fmt.Println("             <version>2.8.9</version>")
+	fmt.Println("         </dependency>")
+	fmt.Println("     </dependencies>")
+	fmt.Println()
+	return nil
+}
+
+// CopyAll copies src to dst. If src is a directory, its contents will be copied
+// recursively.
+func CopyAll(dst string, src string) error {
+	info, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if info.IsDir() {
+		// Recursively copy all files in a directory.
+		files, err := os.ReadDir(src)
+		if err != nil {
+			return fmt.Errorf("read dir: %w", err)
+		}
+		for _, file := range files {
+			name := file.Name()
+			copyerr := CopyAll(filepath.Join(dst, name), filepath.Join(src, name))
+			if copyerr != nil {
+				return copyerr
+			}
+		}
+	} else if info.Mode().IsRegular() {
+		// Copy files by reading and rewriting their contents.  Skip other special files.
+		data, err := os.ReadFile(src)
+		if err != nil {
+			return fmt.Errorf("read file: %w", err)
+		}
+		dstdir := filepath.Dir(dst)
+		if err = os.MkdirAll(dstdir, 0o700); err != nil {
+			return err
+		}
+		if err = os.WriteFile(dst, data, info.Mode()); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func extractModulePath(pkgRef schema.PackageReference) string {
+	var vPath string
+	version := pkgRef.Version()
+	name := pkgRef.Name()
+	if version != nil && version.Major > 1 {
+		vPath = fmt.Sprintf("/v%d", version.Major)
+	}
+
+	// Default to example.com/pulumi-pkg if we have no other information.
+	root := "example.com/pulumi-" + name
+	// But if we have a publisher use that instead, assuming it's from github
+	if pkgRef.Publisher() != "" {
+		root = fmt.Sprintf("github.com/%s/pulumi-%s", pkgRef.Publisher(), name)
+	}
+	// And if we have a repository, use that instead of the publisher
+	if pkgRef.Repository() != "" {
+		url, err := url.Parse(pkgRef.Repository())
+		if err == nil {
+			// If there's any errors parsing the URL ignore it. Else use the host and path as go doesn't expect http://
+			root = url.Host + url.Path
+		}
+	}
+
+	// Support pack sdks write a go mod inside the go folder. Old legacy sdks would manually write a go.mod in the sdk
+	// folder. This happened to mean that sdk/dotnet, sdk/nodejs etc where also considered part of the go sdk module.
+	if pkgRef.SupportPack() {
+		return fmt.Sprintf("%s/sdk/go%s", root, vPath)
+	}
+
+	return fmt.Sprintf("%s/sdk%s", root, vPath)
+}
+
+func NewPluginContext(cwd string) (*plugin.Context, error) {
```

Comment:
I don't think this is worth sharing and it's bloating this PR -- let's put it back.